lib/str_util.c

```diff
@@ -1,10 +1,30 @@
-#include <stdbool.h>
 #include <assert.h>
 #include <string.h>
 #include <fitsio.h>
 
 #include "str_util.h"
 
+/**
+ * Fills a buffer with `ch` up to `max_ch` times.
```

Comment:
Why do I see Sphinx doc syntax here? Probably unnecessary?

Reply:
1) Habit
2) It's readable
3) And secretly makes things like this happen

![Screenshot_20250605_145044](https://github.com/user-attachments/assets/035e3be1-4f1a-45ce-986f-7dcbbf427bb9)